### ISSUE TITLE
4.0.x Connection::PARAM_STR_ARRAY do not work

### ISSUE THREAD

**Issue by OliverSkroblin** [doctrine/dbal#5837]
|    Q        |   A
|------------ | -----
| Version     | 4.0.x

### Support Question
I 
am currently trying out 4.0.x. After fixing some issues in our software, I found out that Connection::PARAM_STR_ARRAY is no more a valid argument for the query builder `setParameter`:

```
$query->setParameter('ids', $ids, Connection::PARAM_STR_ARRAY);
```

I've been looking for an alternative, but haven't found one yet. Is this just missing in the ParameterType enum?

**Comment by derrabus** [comment]
Good catch. Yes, I believe, the type of the `$type` parameter is too narrow here. Adding it to the `ParameterType` enum makes sense, but we should probably do so already in 3.6.

**Comment by derrabus** [comment]
Looking into this a bit more, there's actually a reason why the `*_ARRAY` constants are not part of the `ParameterType` enum. That enum is also used for driver-level parameter binding, but the array expansion happens in the wrapper-level connection already. So we probably better create a second enum for that.

**Comment by derrabus** [comment]
Can you have a look if #5839 works for you?

**Comment by alexander-schranz** [comment]
Currently seeing following error in our CI:

```
Error: Undefined constant Doctrine\DBAL\Connection::ARRAY_PARAM_OFFSET

/home/runner/work/sulu/sulu/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php:1962
/home/runner/work/sulu/sulu/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php:1963
/home/runner/work/sulu/sulu/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php:1879
/home/runner/work/sulu/sulu/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php:915
/home/runner/work/sulu/sulu/vendor/doctrine/orm/lib/Doctrine/ORM/EntityRepository.php:225
/home/runner/work/sulu/sulu/vendor/doctrine/orm/lib/Doctrine/ORM/EntityRepository.php:368
/home/runner/work/sulu/sulu/vendor/doctrine/orm/lib/Doctrine/ORM/EntityRepository.php:272
/home/runner/work/sulu/sulu/src/Sulu/Bundle/ContactBundle/Contact/AccountManager.php:286
```

Think this is related to this issue?

```
  - Locking doctrine/dbal (3.6.x-dev 7e0d352)
  - Locking doctrine/orm (2.15.x-dev ae9fb8c)
```

It did not appear on `3.6.x-dev 63e513c`. So I think it is related to changes in #5838.

---------------

This was fixed in #5842
